pydantic/_internal/_generate_schema.py

```diff
@@ -843,8 +821,11 @@ def _generate_schema_from_property(self, obj: Any, source: Any) -> core_schema.C
             ref = get_ref(schema)
 
         if ref:
-            self.defs.definitions[ref] = schema
-            return core_schema.definition_reference_schema(ref)
+            return self.defs.create_reference_to_schema(schema)
+
+        # TODO
+        # if schema['type'] == 'definition-ref':
+        #     return core_schema.definition_reference_schema(schema_ref=schema['schema_ref'])
```

Comment:
@MarkusSintonen, this is an addition in your PR. Seems like test passes without it. Do you remember why this might be required? I'm assuming this is part of https://github.com/pydantic/pydantic/pull/10655, where `'definition-ref'` schemas are inlined in place, and thus this might some kind of safety copy?